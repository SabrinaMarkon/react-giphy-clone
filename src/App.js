import React from 'react';
import {
  BrowserRouter as Router,
  Route,
  Link,
  Switch,
  Redirect
} from "react-router-dom";
import Dashboard from './components/Dashboard';
import Create from './components/Create';
import Callback from './components/Callback';
import { createBrowserHistory } from 'history';

const App = () => {
  console.log("app component running");
  return (
    <div className="container">
      <Router>
        <Switch>
          <Route exact path="/" component={Dashboard} />
          <Route path="/create" component={Create} />
          <Route path="/callback" component={Callback} />
        </Switch>
      </Router>
    </div>
  );
}

export default App;