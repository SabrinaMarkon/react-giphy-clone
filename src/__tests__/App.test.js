import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup, waitForElement } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import axiosMock from 'axios';
import App from '../App';

afterEach(cleanup);

it('matches the snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

it('fetches and displays the data (gifs)', async () => {
  const { getByAltText } = render(<App />);
  const progressImg = getByAltText('Loading...');
  expect(progressImg).toBeInTheDocument();
});