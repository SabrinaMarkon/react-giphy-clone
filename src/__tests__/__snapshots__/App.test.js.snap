// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div>
      <nav
        class="navbar navbar-inverse"
      >
        <div
          class="container-fluid"
        >
          <div
            class="navbar-header"
          >
            <a
              class="navbar-brand"
              href="/"
            >
              React Giphy Clone
            </a>
          </div>
          <div>
            <ul
              class="nav navbar-nav"
            >
              <li>
                <a
                  href="/"
                >
                  All Gifs
                </a>
              </li>
              <li />
            </ul>
            <ul
              class="navbar navbar-nav navbar-right"
            >
              <li
                style="list-style-type: none;"
              >
                 
              </li>
              <li
                style="list-style-type: none;"
              >
                <button
                  class="btn btn-raised btn-md btn-info"
                  style="margin-right: 20px;"
                  type="button"
                >
                  Login
                </button>
              </li>
            </ul>
          </div>
        </div>
      </nav>
      <div
        class="row"
      >
        <div
          style="text-align: center; padding: 15px 15px 25px 15px;"
        >
          <h1
            style="font-size: 40px; color: rgb(63, 81, 181); margin: 10px 10px 20px 10px;"
          >
            Convert your mp4 or webm video to a small, quality gif!
          </h1>
          <img
            alt="Loading..."
            src="./images/loader.gif"
            style="width: 75px; margin-top: 100px;"
          />
        </div>
        <div />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
